temporian/core/operators/window/moving_sum.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Moving Sum operator."""
+from typing import Optional, List
+
+from temporian.core import operator_lib
+from temporian.core.data.duration import Duration
+from temporian.core.data.event import Event
+from temporian.core.data.feature import Feature
+from temporian.core.data.sampling import Sampling
+from temporian.core.operators.window.base import BaseWindowOperator
+
+
+class MovingSumOperator(BaseWindowOperator):
+    """
+    Window operator to compute the moving sum.
+    """
+
+    @classmethod
+    @property
+    def operator_def_key(cls) -> str:
+        return "MOVING_SUM"
+
+    @property
+    def prefix(self) -> str:
+        return "moving_sum"
+
+    def output_features(
+        self,
+        event: Event,
+        sampling: Sampling,
+    ) -> List[Feature]:
+        features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=f"{self.prefix}_{f.name()}",
+                dtype=f.dtype(),
+                sampling=sampling,
+                creator=self,
+            )
+            for f in event.features()
+        ]
+
+        return features
+
+
+operator_lib.register_operator(MovingSumOperator)
+
+
+def moving_sum(
+    event: Event,
+    window_length: Duration,
+    sampling: Optional[Event] = None,
+) -> Event:
+    """Moving Sum.
+
+    For each sampling, and for each feature independently, returns at time "t"
+    the sum of the feature in the time windows [t-window, t].
+
+    If "sampling" is provided, applies the operator for each timestamps of
+    "sampling". If "sampling" is not provided, apply the operator for each
+    timestamps of "event".
+
+    Missing values are ignored.
+
+    If the window does not contain any values (e.g., all the values are missing,
+    or the window does not contain any sampling), outputs missing values.
```

Comment:
In response to note in description: I would definitely output NaN instead of zero if all the values in the window are NaN (which is also what this docstring states)